test/js/ez-field-edit.js

```diff
@@ -0,0 +1,113 @@
+describe('ez-field-edit', function() {
+    let element;
+    let elementWithMultipleInputs;
+    let elementWithoutLabel;
+
+    beforeEach(function () {
+        element = fixture('BasicTestFixture');
+        elementWithMultipleInputs = fixture('BasicTestFixtureMultipleInputs');
+        elementWithoutLabel = fixture('BasicTestFixtureInputWithoutLabel');
+    });
+
+    it('should be defined', function () {
+        assert.equal(
+            window.customElements.get('ez-field-edit'),
+            element.constructor
+        );
+    });
+
+    describe('properties', function () {
+        describe('`invalid`', function () {
+            it('should default to false', function () {
+                assert.notOk(element.invalid);
+            });
+
+            describe('set', function () {
+                it('should be reflected to an attribute', function () {
+                    element.invalid = true;
+
+                    assert.isTrue(
+                        element.hasAttribute('invalid')
+                    );
+                });
+            });
+        });
+    });
+
+    describe('validity', function () {
+        it('should get invalid', function () {
+            element.querySelector('input').value = 'invalid value';
+            element.dispatchEvent(new CustomEvent('input'));
+            assert.isTrue(
+                element.invalid
+            );
+        });
+
+        it('should get valid', function () {
+            element.invalid = true;
+            element.querySelectorAll('input').value = 42;
+            element.dispatchEvent(new CustomEvent('input'));
+            assert.isFalse(
+                element.invalid
+            );
+        });
+    });
+
+    describe('invalid lowest common parentNode between related input and label', function () {
```

Comment:
the test outline is not super clear, it seems like some cases are not tested or mixed.
I would write something like:

```js
describe('`ez-invalid-input` class', function () {
    describe('unique input', function () {
        describe('with label', function () {
            it('should be set on lowest common ancestor');
            it('should be remove from lowest common ancestor');

            describe('label deeper than input', function () {
                it('should be set on lowest common ancestor');
                it('should be remove from lowest common ancestor');
            });

            describe('input deeper than label', function () {
                it('should be set on lowest common ancestor');
                it('should be remove from lowest common ancestor');
            });
        });

        describe('without label', function () {
             it('should be set on parent node');
             it('should be remove from parent node');
        });
    });

    describe('multiple inputs', function () {
        describe('with label', function () {
            it('should be set on lowest common ancestor');
            it('should be remove from lowest common ancestor');

            describe('label deeper than input', function () {
                it('should be set on lowest common ancestor');
                it('should be remove from lowest common ancestor');
            });

            describe('input deeper than label', function () {
                it('should be set on lowest common ancestor');
                it('should be remove from lowest common ancestor');
            });
        });

        describe('without label', function () {
             it('should be set on parent node');
             it('should be remove from parent node');
        });
    });
});
```